app/src/context/slices/projectSlice.ts

```diff
@@ -0,0 +1,1986 @@
+        metadata: payload.metadata,
+        uiSpecificationId: compiledSpecId,
+        rawUiSpecification: payload.rawUiSpecification,
+        lastUpdated: payload.lastUpdated,
+        createdAt: payload.createdAt,
+      };
+    },
+
+    /**
+     * Activates an existing project.
+     *
+     * This involves
+     *
+     * - creating local pouch DB which stores the data synced from the remote (and new records)
+     * - creating the remote pouch DB which is a connection point to the remote data-database
+     * - creating the sync object which performs the synchronisation between the two databases
+     * - registering the above non-serialisable objects into databaseService
+     * - marking the project as activated and updating store state
+     *
+     */
+    activateProject: (
+      state,
+      action: PayloadAction<ProjectIdentity & DatabaseAuth>
+    ) => {
+      const payload = action.payload;
+
+      // Check the server exists
+      const server = serverById(state, payload.serverId);
+      if (!server) {
+        // abort
+        throw new Error(
+          `You cannot activate a project for a server which does not exist. Server ID: ${payload.serverId}. Project ID: ${payload.projectId}`
+        );
+      }
+
+      // Check the couch DB url has been populated
+      if (!server.couchDbUrl) {
+        // abort
+        throw new Error(
+          `Cannot activate when we don't know the couchDBUrl. Server ID: ${payload.serverId}. Project ID: ${payload.projectId}`
+        );
+      }
+
+      // check the project exists
+      const project = projectByIdentity(state, payload);
+      if (!project) {
+        // abort
+        throw new Error(
+          `You cannot activate a project which does not exist. Server ID: ${payload.serverId}. Project ID: ${payload.projectId}`
+        );
+      }
+
+      // check it's not already active
+      if (project.isActivated) {
+        throw new Error(
+          `You cannot activate a project which is already active. Server ID: ${payload.serverId}. Project ID: ${payload.projectId}`
+        );
+      }
+
+      // build the connection info
+      const connectionConfiguration: DatabaseConnectionConfig = {
+        // push in the specified jwt
+        jwtToken: payload.jwtToken,
+        couchUrl: server.couchDbUrl,
+        databaseName: getRemoteDatabaseNameFromId({
+          projectId: project.projectId,
+        }),
+      };
+
+      // creates and/or links to the local data database
+      const localDatabaseId = buildPouchIdentifier({
+        projectId: payload.projectId,
+        serverId: payload.serverId,
+      });
+      const localDb = createLocalPouchDatabase<ProjectDataObject>({
+        id: localDatabaseId,
+      });
+      databaseService.registerLocalDatabase(localDatabaseId, localDb);
+
+      // creates the remote database (pouch remote)
+      const {db: remoteDb, id: remoteDbId} =
+        createRemotePouchDbFromConnectionInfo<ProjectDataObject>(
```

Comment:
Danger!!! What if this is the only copy, I'd be happier if we didn't delete it here or only deleted if we had assurance that the data is safe. Can we just mark this project as deactivated and keep the database until such time as we have a watertight check on the safety of the data?